src/Raven.Client/ServerWide/DatabaseTopology.cs

```diff
@@ -192,6 +192,36 @@ internal void ReorderMembers()
         }
 
 
+        internal bool TryUpdateByPriorityOrder()
+        {
+            if (IsReorderNeeded() == false)
+                return false;
+
+            var originalOrder = new List<string>(Members);
+
+            ReorderMembers();
+            if (IsReorderNeeded() == false)
+                return true;
+
+            // we fail to ensure the order, so let's revert to the original and continue.
+            Members = originalOrder;
+            return false;
+        }
+        private bool IsReorderNeeded()
```

Comment:
I don't understand the logic here. I think that you are checking to see if the members list match the priority order, but don't you need the inverse check in the for ?